### PR TITLE
Adjust indentation to two spaces

### PR DESCRIPTION
**Pull request by joschmitt**
... since it was such a huge success in Nemo and there are not many (active) pull requests open right now.

I can of course easily update this, if something else wants to get merged first.

### CODE REVIEW

File: test/generic/ModuleHomomorphism-test.jl [hunk 1]
```diff
@@ -1,197 +1,197 @@
...
 @testset "Generic.ModuleHomomorphism.printing" begin
   M = free_module(ZZ, 2)
   f = ModuleHomomorphism(M, M, matrix(ZZ, 2, 2, [1, 2, 3, 4]))
   str = """
-        Module homomorphism
-          from free module of rank 2 over integers
-          to free module of rank 2 over integers"""
+  Module homomorphism
+  from free module of rank 2 over integers
+  to free module of rank 2 over integers"""
```
Comment (lgoettgens) [inline test/generic/ModuleHomomorphism-test.jl:156]: This change is wrong!

Comment (joschmitt) [reply test/generic/ModuleHomomorphism-test.jl:156]: Yes, the tests are also failing. I look at it later.

**Comment by lgoettgens** [comment]
https://github.com/Nemocas/AbstractAlgebra.jl/pull/1704 is the only larger PR, but only adds lines so it should be fine

File: test/generic/ModuleHomomorphism-test.jl (hunk 1 above)
Comment (joschmitt) [reply test/generic/ModuleHomomorphism-test.jl]: Should be fine now.

**Comment by lgoettgens** [comment]
Any objections to just merging this @thofma @fingolfin ?  There are no open PRs with painful conflicts.

File: src/AbsMSeries.jl [hunk 2]
```diff
@@ -12,18 +12,18 @@
...
-    prec = [i == ind ? exps[i] : R.prec_max[i] for i in 1:length(exps)]
-    return R(parent(p)(), prec)
+  if iszero(a)
+    return deepcopy(a)
+  end
+  R = parent(a)
+  p = poly(a)
+  v = vars(p)
+  (length(v) != 1 || length(p) != 1 || !isone(leading_coefficient(p))) &&
+  error("Not a pure power in O()")
```
Comment (thofma) [inline src/AbsMSeries.jl:22]: This does not look.

**Comment by thofma** [comment]
What is up with all the 
```
something_very_long = [
  1, 2, 3,
  4, 5, 6
```
being changed to
```
something_very_long = [
                                          1, 2, 3,
                                          4, 5, 6
```
and similar for all the other parantheses.

**Comment by joschmitt** [comment]
> What is up with all the
> 
> ```
> something_very_long = [
>   1, 2, 3,
>   4, 5, 6
> ```
> 
> being changed to
> 
> ```
> something_very_long = [
>                                           1, 2, 3,
>                                           4, 5, 6
> ```
> 
> and similar for all the other parantheses.

Can you point to an example? I'm not sure what you mean.

File: docs/make.jl [hunk 3]
```diff
@@ -8,98 +8,98 @@ using Documenter, AbstractAlgebra
 
 makedocs(
          format = Documenter.HTML(;
-            size_threshold_warn = 204800,
-            size_threshold = 409600,
-         ),
+                                  size_threshold_warn = 204800,
```
Comment (thofma) [inline docs/make.jl:11]: This is what I mean. I messes again with carefully introduced line breaks.

(Sorry, I did not get the indentation right in my github comment.)

File: docs/make.jl [hunk 4]
```diff
@@ -8,98 +8,98 @@ using Documenter, AbstractAlgebra
...
-                 "matrix_interface.md",
-                 "map_interface.md",
-                 "rand.md",
-                 "linear_solving.md",
-             ],
-         ]
-)
+                     "index.md",
+                     "Fundamental interface of AbstractAlgebra.jl" => [
+                                                                       "types.md",
```
Comment (thofma) [inline docs/make.jl:22]: this is awful

File: src/AbsMSeries.jl (hunk 2 above)
Comment (joschmitt) [reply src/AbsMSeries.jl]: I made an `if` out of this. It took me a moment to understand what these `||` and `&&` do.

File: src/AbsMSeries.jl [hunk 5]
```diff
@@ -154,53 +155,53 @@ end
...
 
 function RandomExtensions.make(S::MSeriesRing,
-                                      term_range::AbstractUnitRange{Int}, vs...)
-   R = base_ring(S)
-   if length(vs) == 1 && elem_type(R) == Random.gentype(vs[1])
-      Make(S, term_range, vs[1])
-   else
-      Make(S, term_range, make(R, vs...))
-   end
+    term_range::AbstractUnitRange{Int}, vs...)
```
Comment (thofma) [inline src/AbsMSeries.jl:158]: here it doesn't do it, but it does the opposite?

File: src/AbsMSeries.jl [hunk 6]
```diff
@@ -154,53 +155,53 @@ end
...
-   if S.weighted_prec == -1
-      prec = max_precision(S)
-      for i = 1:rand(rng, term_range)
-         term = S(1)
-         for j = 1:length(g)
-            term *= g[j]^rand(rng, 0:prec[j])
-         end
-         term *= rand(rng, v)
-         f += term
+                                                           <:RingElement, <:MSeriesRing, <:AbstractUnitRange{Int}}})
```
Comment (thofma) [inline src/AbsMSeries.jl:168]: there is no need for a line break if one does it like this, but why change it?

File: src/AbsMSeries.jl (hunk 5 above)
Comment (joschmitt) [reply src/AbsMSeries.jl:158]: I don't know what the internal logic of the Julia-Vim plugin does. Maybe it reacts to the number of open brackets?

**Comment by joschmitt** [comment]
If we are generally happy with this, I can try to remove some unwanted changes by hand. (Starting with resetting `docs/make.jl`.)

**Comment by thofma** [comment]
I only checked the first 100 lines. I have 49700 to go.

**Comment by joschmitt** [comment]
> I only checked the first 100 lines. I have 49700 to go.

Yes, that's also my problem.

**Comment by joschmitt** [comment]
Before I put more work into this: What's the idea? Status quo? This pull request (most lines are correctly indented and a few aren't)? We hire somebody to hand-indent every line? ...?

**Comment by thofma** [comment]
Status quo it is. The inconsistent indentation does not effect readability. On the other hand, the changes here do not add anything for readability, but they are actually making it worse. See the
```
a || b || c &&
  error()
```
change to
```
a || b || c &&
error()
```
And these are only the ones I found. I have not checked the other 49000 lines of code. I would say, we just change it when we work on it.

**Comment by joschmitt** [comment]
Okay!

**Pull request closed without merging**